From roel.kluin@gmail.com  Tue Jan  5 14:37:31 2010
From: Roel Kluin <roel.kluin@gmail.com>
Date: Sat, 26 Dec 2009 20:00:47 +0100
Subject: Staging: rt2860: test off by one in RtmpAsicSendCommandToMcu()
To: Greg Kroah-Hartman <gregkh@suse.de>, devel@driverdev.osuosl.org,  Andrew Morton <akpm@linux-foundation.org>, bzolnier@gmail.com
Message-ID: <4B365D5F.70105@gmail.com>


`i' reaches 101 after the loop, so if it was 100 then it succeeded in
the last iteration. This is probably unlikely to cause problems.

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2860/common/rtmp_mcu.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/rt2860/common/rtmp_mcu.c
+++ b/drivers/staging/rt2860/common/rtmp_mcu.c
@@ -221,7 +221,7 @@ int RtmpAsicSendCommandToMcu(struct rt_r
 				 ("AsicSendCommanToMcu::Mail box is busy\n"));
 		} while (i++ < 100);
 
-		if (i >= 100) {
+		if (i > 100) {
 			DBGPRINT_ERR(("H2M_MAILBOX still hold by MCU. command fail\n"));
 			return FALSE;
 		}
